qiskit_ibm_runtime/runtime_job.py

```diff
@@ -245,7 +245,7 @@ def cancel(self) -> None:
         try:
             self._api_client.job_cancel(self.job_id())
         except RequestsApiError as ex:
-            if ex.status_code == 409:
+            if ex.status_code in (409, 204):
```

Comment:
Thanks for taking care of this @robotAstray! I think this change is unnecessary because if the status code was 204 there would be no exception caught.

Reply:
@kt474 Ok Thank you! that makes sense. I will add the changes